.github/workflows/pkgci_regression_test_amdgpu_vulkan.yml

```diff
@@ -40,12 +40,33 @@ jobs:
           path: ${{ env.PACKAGE_DOWNLOAD_DIR }}
       - name: Setup venv
         run: |
-          ./build_tools/pkgci/setup_venv.py $VENV_DIR \
+          ./build_tools/pkgci/setup_venv.py ${VENV_DIR} \
             --artifact-path=${PACKAGE_DOWNLOAD_DIR} \
             --fetch-gh-workflow=${{ inputs.artifact_run_id }}
-      - name: Run Tests
+
+      # In-tree tests
+      - name: Run experimental/regression_suite tests
         run: |
-          source $VENV_DIR/bin/activate
+          source ${VENV_DIR}/bin/activate
           pytest \
             -rA -s -m "plat_rdna3_vulkan and presubmit" \
             experimental/regression_suite
+
+      # Out of tree tests
+      - name: Checking out external TestSuite repository
+        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # v3.5.0
+        with:
+          repository: ScottTodd/SHARK-TestSuite  # replace with `nod-ai/SHARK-TestSuite`
+          ref: testsuite-pytest  # replace with a pinned commit on `main`
+          path: SHARK-TestSuite
+          submodules: false
+      - name: Installing external TestSuite Python requirements
+        run: |
+          source ${VENV_DIR}/bin/activate
+          python -m pip install -r SHARK-TestSuite/iree_tests/requirements.txt
+      - name: Run external TestSuite tests
+        env:
+          IREE_TEST_CONFIG_FILES: experimental/regression_suite/external_test_suite/config_gpu_vulkan.json
+        run: |
+          source ${VENV_DIR}/bin/activate
+          pytest SHARK-TestSuite/iree_tests -n auto -rpfE
```

Comment:
Bah. Compiler crashes for Vulkan and ROCm that aren't even interacting nicely with pytest:
* https://github.com/openxla/iree/actions/runs/8072456690/job/22054484578?pr=16589#step:9:457
* https://github.com/openxla/iree/actions/runs/8072261795/job/22053870186#step:9:29

even with changes removed? Huh?? something wrong in the env / with the multithreading? https://github.com/openxla/iree/actions/runs/8072456690/job/22054484855?pr=16589#step:6:42

Reply:
this is during compilation it looks like? that's really weird!
(96 parallel tests creating vulkan/rocm devices seems like it'd be a stress test, but the compiler should be fine)

Reply:
Yeah I turned off the ROCm tests while I wanted to investigate on my machine... but it looks like something still needs debugging even with Vulkan. I can turn down/off the parallelism on the GPU machines to see if that helps at least isolate what is failing.

Reply:
It was (up to) 32 Vulkan devices yeah... Dropped down to 4 and the tests still only took 30 seconds but didn't have any crashes. HMMMM... will monitor carefully after merge I guess.

Reply:
Seems stable with 4 threads instead of 32.